### PR TITLE
Set MaxMetaspaceSize to 512m to fix Dokka Gradle build step

### PR DESCRIPTION
**Pull request by talklittle**
Without this configuration, the `:core:dokkaJavadoc` Gradle build step was failing/hanging with error:

```
Exception in thread "DefaultDispatcher-worker-7" java.lang.OutOfMemoryError: Metaspace
The Daemon will expire after the build after running out of JVM Metaspace.
The project memory settings are likely not configured or are configured to an insufficient value.
The daemon will restart for the next build, which may increase subsequent build times.
These settings can be adjusted by setting 'org.gradle.jvmargs' in 'gradle.properties'.
The currently configured max heap space is '512 MiB' and the configured max metaspace is '256 MiB'.
For more information on how to set these values, visit the user guide at https://docs.gradle.org/8.0/userguide/build_environment.html#configuring_jvm_memory
To disable this warning, set 'org.gradle.daemon.performance.disable-logging=true'.
Daemon will be stopped at the end of the build after running out of JVM Metaspace
```

For whatever reason the build worked on `master` but not current `dev` branch.

Specifically I was running `$ ./gradlew :core:publishToMavenLocal`

### CODE REVIEW

File: gradle.properties [hunk 1]
```diff
@@ -20,4 +21,4 @@ android.defaults.buildfeatures.buildconfig=true
 android.enableJetifier=true
 android.nonFinalResIds=false
 android.nonTransitiveRClass=false
-android.useAndroidX=true
\ No newline at end of file
+android.useAndroidX=true
```
Comment (PierfrancescoSoffritti) [inline gradle.properties:24]: Remove unnecessary change

**Comment by talklittle** [comment]
Thanks, updated.

**Comment by talklittle** [comment]
Thanks for all the feedback on these pull requests, and for the library!
